From 86c66656c12f6359d8905cb241b139df8106a12a Mon Sep 17 00:00:00 2001
From: Marek Marczykowski <marmarek@invisiblethingslab.com>
Date: Sun, 14 Apr 2013 04:20:07 +0200
Subject: [PATCH 10/15] libxl: initialize device structures
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
Organization: Invisible Things Lab
Cc: Marek Marczykowski-Górecki <marmarek@invisiblethingslab.com>

Do not leave uninitialized variables, not all parameters are set in
libxlMake*.

Signed-off-by: Marek Marczykowski-Górecki <marmarek@invisiblethingslab.com>
---
 src/libxl/libxl_conf.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/libxl/libxl_conf.c b/src/libxl/libxl_conf.c
index d903870..c40336e 100644
--- a/src/libxl/libxl_conf.c
+++ b/src/libxl/libxl_conf.c
@@ -898,6 +898,8 @@ int
 libxlMakePci(virDomainHostdevDefPtr l_hostdev,
              libxl_device_pci *x_pci)
 {
+    libxl_device_pci_init(x_pci);
+
     if (l_hostdev->mode != VIR_DOMAIN_HOSTDEV_MODE_SUBSYS) {
         virReportError(VIR_ERR_CONFIG_UNSUPPORTED, "%s", _("This driver supports only subsystem host device mode (PCI devices only)"));
         return -1;
@@ -1004,6 +1006,8 @@ libxlBuildDomainConfig(libxlDriverPrivatePtr driver,
 
     libxl_domain_config_init(d_config);
 
+    libxl_domain_config_init(d_config);
+
     if (libxlMakeDomCreateInfo(driver, def, &d_config->c_info) < 0)
         return -1;
 
-- 
1.8.1.4

